Refactor logging mechanism for printStackTrace

shahidtamboli opened the pull request:
Initially we had used <code>printStackTrace</code> everywhere for printing error logs. 
We have now changed that and are using android's logging to print stack traces when the application is in debug mode.

@nasthagiri @rohan-dhamal-clarice -  Please review the changes

cc - @aleffert

aleffert commented on android/source/VideoLocker/src/org/edx/mobile/logger/OEXLogger.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Created by shahid on 22/1/15.
+ */
+public class OEXLogger {
```
You don't need to call things ``OEX`` on android, because Java has a built in namespace mechanism. That's only necessary in Objective-C where there's a single flat namespace.

aleffert commented:
Great! Remove the OEX thing and push it :+1:

rohan-dhamal-clarice commented on android/source/VideoLocker/src/org/edx/mobile/view/SplashActivity.java:
```diff
@@ -26,9 +25,6 @@ protected void onCreate(Bundle savedInstanceState) {
         overridePendingTransition(R.anim.slide_in_from_right, R.anim.slide_out_to_left);
 
-        // initialize log util
-        LogUtil.init(this);
```
@shahidtamboli We need to initialize logging module from <code>MainApplication</code>. Without this, logs will also be printed in release mode.

rohan-dhamal-clarice commented on android/source/VideoLocker/src/org/edx/mobile/logger/LogUtil.java:
```diff
@@ -0,0 +1,74 @@
+     * @param context - current context passed
+     */
+    public static void init(Context context) {
+        boolean isDebuggable =  ( 0 != ( context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE));
```
@shahidtamboli To be more specific, this line should use <code>ApplicationInfo.FLAG_DEBUGGABLE ==</code> instead of <code>0 !=</code>.

shahidtamboli commented:
@aleffert - I have renamed the class to from <code>OEXLogger</code> to <code>Logger</code>.

@rohan-dhamal-clarice - I have updated the code as per your suggestion and have initialized the logging module in <code>MainApplication</code>.
Please review.

rohan-dhamal-clarice commented:
:+1:
